dask/dot.py

```diff
@@ -273,7 +273,10 @@ def dot_graph(dsk, filename="mydask", format=None, **kwargs):
 
 def graphviz_to_file(g, filename, format):
     fmts = [".png", ".pdf", ".dot", ".svg", ".jpeg", ".jpg"]
-    if format is None and any(filename.lower().endswith(fmt) for fmt in fmts):
+
+    if format is None and filename is None:
+        pass
+    elif format is None and any(filename.lower().endswith(fmt) for fmt in fmts):
```

Comment:
Actually, these three lines could be condensed, either
```
if format is None and filename is not None and any(filename.lower().endswith(fmt) for fmt in fmts):
```
or
```
if format is None and any((filename or "").lower().endswith(fmt) for fmt in fmts):
```
The second is shorter, the first is clearer.

Reply:
I added the first one! Even though it's longer, using `black` formatting, it looks very clean.

```py
if (
    format is None
    and filename is not None
    and any(filename.lower().endswith(fmt) for fmt in fmts)
):
    filename, format = os.path.splitext(filename)
    format = format[1:].lower()
```